openmdao/drivers/tests/test_pyoptsparse_driver.py

```diff
@@ -104,6 +106,38 @@ def compute_partials(self, inputs, partials):
 
         partials['y', 'x'] = 2.0*x - 6.0
 
+@unittest.skipIf(OPT is None or OPTIMIZER is None, "only run if pyoptsparse is installed.")
+@unittest.skipUnless(MPI, "MPI is required.")
+class TestMPIScatter(unittest.TestCase):
+    N_PROCS = 2
+
+    def test_design_vars_on_all_procs(self):
+
+        prob = om.Problem()
+        model = prob.model
+
+        model.add_subsystem('p1', om.IndepVarComp('x', 50.0), promotes=['*'])
+        model.add_subsystem('p2', om.IndepVarComp('y', 50.0), promotes=['*'])
+        model.add_subsystem('comp', Paraboloid(), promotes=['*'])
+        model.add_subsystem('con', om.ExecComp('c = - x + y'), promotes=['*'])
+
+        prob.set_solver_print(level=0)
+
+        prob.driver = pyOptSparseDriver(optimizer=OPTIMIZER, print_results=False)
+        if OPTIMIZER == 'SLSQP':
+            prob.driver.opt_settings['ACC'] = 1e-9
+
+        model.add_design_var('x', lower=-50.0, upper=50.0)
+        model.add_design_var('y', lower=-50.0, upper=50.0)
+        model.add_objective('f_xy')
+        model.add_constraint('c', upper=-15.0)
+
+        prob.setup()
+        prob.run_driver()
+
+        design_vars = MPI.COMM_WORLD.allgather([prob['x'], prob['y'], prob['c'], prob['f_xy']], root=0)
```

Comment:
Don't need to do an allgather in the test. Just assert that each variable(e.g., prob['x]) equals the static value (e.g., 7.1666667). The test runs on both procs, so such a test would verify that the correct value of the desvars, objs, and cons have been propagated to each proc.

Reply:
Updated